pkg/authz/oauth/oktaprovider.go

```diff
@@ -14,23 +22,49 @@ func (i *okta) getAuthorizationHeaderPrefix() string {
 	return oktaAuthHeaderPrefix
 }
 
-func (i *okta) preProcessClientRequest(clientRequest *clientMetadata) {
-	if len(clientRequest.extraProperties) == 0 {
-		clientRequest.extraProperties = make(map[string]string)
+// validateExtraProperties validates Okta-specific extra properties and sets defaults
+func (i *okta) validateExtraProperties(extraProps map[string]interface{}) error {
+	pkceRequired, _ := extraProps[oktaPKCERequired].(bool)
+	appType, hasAppType := extraProps[oktaApplicationType].(string)
+
+	// If application_type is already set, validate it
+	if hasAppType && appType != "" {
+		if pkceRequired && appType != oktaAppTypeBrowser {
+			return fmt.Errorf("when %s is true, %s must be '%s' or unset, got '%s'",
+				oktaPKCERequired, oktaApplicationType, oktaAppTypeBrowser, appType)
+		}
+		return nil
+	}
```

Comment:
Is the return nil here needed? Would this be what is needed?
```
if pkceRequired && appType != oktaAppTypeBrowser {
		return fmt.Errorf("when %s is true, %s must be '%s' or unset, got '%s'",
				oktaPKCERequired, oktaApplicationType, oktaAppTypeBrowser, appType)
		}
}
```

Reply:
return nil is not, but i believe i still need to check appType != ""

if pkceRequired && appType != "" && appType != oktaAppTypeBrowser {